### PR TITLE
Add author image to blog posts

### PR DESCRIPTION
**Pull request by joaoGabriel55**
## Description

Instead of the initials of each author, now use the GitHub image for each author

## Validation

image

image

image

## Related Issues

Related to #6573

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by AugustinMauroy — CHANGES_REQUESTED** [review]
At moment this not solve issue. you just remove initial and put nodejs org profile picture.

So like say on the issue. My opinion it's to use front matter to have profile picture and if it's nodejs project -> use nodejs org as username. And make small work on markdown do verify if major of author is right username.

**Comment by ovflowd** [comment]
Blog Posts do not have any username at the moment, so this isn't really fixing any issue unfortunately.

We should probably open an issue to keep track of the problem and actually come with a long-term solution.

Unfortunately, not all blog authors have a GitHub account... And we have blog posts from back 2011.

Here's a full list of all blog authors and usernames when available: https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml

Feel free to make a script to replace the existing names to usernames on all blog posts, and then update the script that creates release blog posts to use username instead of name ;)

Then this solution is viable. But we still need their names, so 🤷 if you're feeling like spending time into this, go ahead! It's appreciated

**Comment by AugustinMauroy** [comment]
> We should probably open an issue to keep track of the problem and actually come with a long-term solution.

Could you do it ?

> Here's a full list of all blog authors and usernames when available:  https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml

I thinks it's will easier to follow you proposal. But use JSON instead of yaml it's can be link to `next.json.mjs`.

**Comment by ovflowd** [comment]
> > We should probably open an issue to keep track of the problem and actually come with a long-term solution.
> 
> Could you do it ?
> 
> > Here's a full list of all blog authors and usernames when available:  [nodejs/nodejs.dev@`main`/src/data/blog/authors.yaml](https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml?rgh-link-date=2024-03-29T07%3A45%3A47Z)
> 
> I thinks it's will easier to follow you proposal. But use JSON instead of yaml it's can be link to `next.json.mjs`.

Not everything needs to be done by me 🙈

**Comment by joaoGabriel55** [comment]
> Blog Posts do not have any username at the moment, so this isn't really fixing any issue unfortunately.
> 
> We should probably open an issue to keep track of the problem and actually come with a long-term solution.
> 
> Unfortunately, not all blog authors have a GitHub account... And we have blog posts from back 2011.
> 
> Here's a full list of all blog authors and usernames when available: https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml
> 
> Feel free to make a script to replace the existing names to usernames on all blog posts, and then update the script that creates release blog posts to use username instead of name ;)
> 
> Then this solution is viable. But we still need their names, so 🤷 if you're feeling like spending time into this, go ahead! It's appreciated

Maybe we could still use the avatar with initial letters if the blog post does not have a username field otherwise we can use this new approach.

**Comment by AugustinMauroy** [comment]
basically, there is https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml which is a list that transfers the string that is in authors FrontMatter into a link. All you need to do is put it in JSON instead of YAML.

**Comment by joaoGabriel55** [comment]
That would be my proposal:
```ts
export const mapAuthorToCardAuthors = (author: string, username?: string) => {
  const authors = author.split(/, | and |;| by /i);
  const usernames = username?.split(/, | and |;| by /i) || [];

  return authors.map((fullName, index) => {
    let src = `https://ui-avatars.com/api/?name=${fullName}`;

    if (username) src = getGitHubAvatarUrl(usernames[index]);

    return { fullName, src };
  });
};
```

**Comment by joaoGabriel55** [comment]
> basically, there is https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml which is a list that transfers the string that is in authors FrontMatter into a link. All you need to do is put it in JSON instead of YAML.

I see

**Comment by joaoGabriel55** [comment]
> basically, there is https://github.com/nodejs/nodejs.dev/blob/main/src/data/blog/authors.yaml which is a list that transfers the string that is in authors FrontMatter into a link. All you need to do is put it in JSON instead of YAML.

So just to confirm @AugustinMauroy, in the `blogUtils.ts` we will need to map the `fullname` of the author with the author name in the new JSON file generated by the `authors.yaml` file, right?

**Comment by AugustinMauroy** [comment]
Oh cool !
note:
- "the node.js project" have 404 GitHub image
- "Brian Muenzenmeyer" is missing on json https://github.com/bmuenzenmeyer
- "Marco Ippolito" is missing on json https://github.com/marco-ippolito
- "Ulises Gascón" is missing on json https://github.com/UlisesGascon
-  "Juan José Arboleda" is on JSON but there are also "Juan José" on frontMatter so you should do something to merge it
- "Claudio Wunder" is missing on json https://github.com/ovflowd
- "Matt Cowley" is missing on json https://github.com/MattIPv4
- "ryandahl" isn't working

File: authors.json [hunk 1]
```diff
@@ -0,0 +1,217 @@
+{
+  "Manish Kumar": {
+    "id": "manishprivet",
+    "name": "Manish Kumar",
+    "website": "https://github.com/manishprivet"
+  },
```
Comment (AugustinMauroy) [inline authors.json]:
```suggestion
```

Manish isn't a blog author

**Comment by joaoGabriel55** [comment]
> Oh cool ! note:
> 
> * "the node.js project" have 404 GitHub image
> * "Brian Muenzenmeyer" is missing on json https://github.com/bmuenzenmeyer
> * "Marco Ippolito" is missing on json https://github.com/marco-ippolito
> * "Ulises Gascón" is missing on json https://github.com/UlisesGascon
> * "Juan José Arboleda" is on JSON but there are also "Juan José" on frontMatter so you should do something to merge it
> * "Claudio Wunder" is missing on json https://github.com/ovflowd
> * "Matt Cowley" is missing on json https://github.com/MattIPv4
> * "ryandahl" isn't working

Is the Ryan Dahl who has the GitHub username 'ry' the same person as 'ryandahl'?

**Review by Cloudstream-io — COMMENTED** [review]
Hi looks good

**Comment by ovflowd** [comment]
> > Oh cool ! note:
> 
> > 
> 
> > * "the node.js project" have 404 GitHub image
> 
> > * "Brian Muenzenmeyer" is missing on json https://github.com/bmuenzenmeyer
> 
> > * "Marco Ippolito" is missing on json https://github.com/marco-ippolito
> 
> > * "Ulises Gascón" is missing on json https://github.com/UlisesGascon
> 
> > * "Juan José Arboleda" is on JSON but there are also "Juan José" on frontMatter so you should do something to merge it
> 
> > * "Claudio Wunder" is missing on json https://github.com/ovflowd
> 
> > * "Matt Cowley" is missing on json https://github.com/MattIPv4
> 
> > * "ryandahl" isn't working
> 
> 
> 
> Is the Ryan Dahl who has the GitHub username 'ry' the same person as 'ryandahl'?

Yes!

**Review by AugustinMauroy — APPROVED** [review]
LGMT !! Thanks for your contribution

**Review by canerakdas — APPROVED** [review]
LGTM 🚀 🐢 

This is not related to this PR, but at some point, it may be more efficient to use the `next/image` component together with the `RadixAvatar.Image` component

**Review by bmuenzenmeyer — APPROVED** [review]
cool stuff - thanks!

**Pull request merged**